### DISCUSSION TITLE
Assistant refuse to call remote functions on remote server.

### DISCUSSION THREAD

**Discussion by Alex-Karmazin** [danny-avila/LibreChat#2856] [Troubleshooting]
### What happened?

I try to migrate my assistant from open ai to LibreChat. While on my PC it works fine with docker installation, it refuses to work on linux server with all the same configuration. It chat with LLM but when it have to make function call it returns error message. "Something went wrong. Here's the specific error message we encountered: 400 Expected tool outputs for call_ids ['call_KG0no7fSyvT7mIKl6ahuKuGT'], got []" I checked on API side and it appears there is  no even request to the server.
I attach log files from my pc where everething work fine and server where I got this error.
[server_debug-2024-05-23.log](https://github.com/danny-avila/LibreChat/files/15437691/server_debug-2024-05-23.log)
[server_error-2024-05-23.log](https://github.com/danny-avila/LibreChat/files/15437717/server_error-2024-05-23.log)
[pc_debug-2024-05-23.log](https://github.com/danny-avila/LibreChat/files/15437738/pc_debug-2024-05-23.log)
[pc_error-2024-05-23.log](https://github.com/danny-avila/LibreChat/files/15437740/pc_error-2024-05-23.log)

### Steps to Reproduce

1 Install LibreChat with docker compose on your PC and make sure assistant working fine with calling functions.
2 Install LibreChat with docker on remote linux server and make sure assistant return error message while try to call remote API function.

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### Screenshots

![Error message](https://github.com/danny-avila/LibreChat/assets/95565703/49055fc6-c210-4605-a485-b8dba7c19a2c)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
You need to share databases for expected functionality with API Actions across instances.

**Reply by danny-avila** [reply]
Use Mongo Atlas with the same URI between the two instances if this is desired.

**Comment by Alex-Karmazin** [comment]
I make workaround with custom endpoint. And at first it causes same behavior, but I figured out that it is because I use localhost instead of host.docker.internal So if someone have same issue please try to check your localhost addresses if you use docker. But with assistants, even change URLs to host.docker.internal did not help. But it is definitely a problem related to docker and not to LibreChat itself.
